src/test/scala/org/labrad/ServerTest.scala

```diff
@@ -234,6 +235,19 @@ class ServerTest extends FunSuite with AsyncAssertions {
     assert(keys.contains("a"))
     assert(!keys2.contains("a"))
   }
+
+  test("server gets context expiration message when client disconnects") { fix =>
+    val ctx = withClient(fix.host, fix.port, fix.password) { client2 =>
+      val ctx = client2.newContext.copy(high = client2.id)
+      await(client2.send("Scala Test Server", ctx, "Set" -> Cluster(Str("a"), Str("test"))))
+      ctx
+    }
+    Thread.sleep(1000)
+    val result = await(fix.client.send("Scala Test Server", ctx, "Keys" -> Data.NONE))
+    val keys = result(0).get[Seq[String]]
+    assert(keys == Seq("Test"))  // should just be the test key
+  }
+
 }
 
```

Comment:
see the updated code; the expired context gets passed through a queue (similar to the pylabrad test) so the code can check that the right context got expired.